be/src/olap/rowset/alpha_rowset.cpp

```diff
@@ -300,7 +300,7 @@ OLAPStatus AlphaRowset::init() {
 
         if (segment_group_meta.zone_maps_size() != 0) {
             size_t zone_maps_size = segment_group_meta.zone_maps_size();
-            size_t num_key_columns = _schema->num_key_columns();
+            size_t num_key_columns = _schema->keys_type() == KeysType::DUP_KEYS ? _schema->num_columns() : _schema->num_key_columns();
```

Comment:
I think we should change the num_key_columns name.
num_key_columns  will be ambiguous.

Reply:
it better to submit anthor pr to fix this